pkg/yang/ast.go

```diff
@@ -30,12 +30,16 @@ import (
 	"strings"
 )
 
+func init() {
+	initTypes(reflect.TypeOf(&meta{}))
```

Comment:
Add a comment here just to mention what we're doing here - since I guess you're going to move this out of `init` when we have a context within which modules are generated?

Reply:
Added comment, although I'm not planning on moving it out of `init`. Please see OP.

Reply:
maybe I'm missing something on the design here.

when we have a context for the generation of modules, I presume all the context for any generation will be within the `Modules` parent (or whatever you call `Parse()` on). is that incorrect? I'm struggling a little to see why we need to initialise a global in `init()` if we're not planning to have globals, but I expect I'm missing something here.

Reply:
Under the proposal by this PR we will have globals. It's just that the globals will be constant. i.e. variables that won't ever change, namely the three global variables mentioned in the OP.

The alternative would be to store these variables in `Modules` as well, but I don't see the need because as long as we don't have another type hierarchy (currently rooted at `meta`), for example if we wanted to be able to parse YANG1.1 separately, then there is no need to initialize these global variables again and again each time we instantiate a `Modules` object. Let me know if this made sense.

Reply:
SGTM, makes entire sense.